analyzer/src/main/kotlin/curation/FilePackageCurationProvider.kt

```diff
@@ -43,7 +43,7 @@ class FilePackageCurationProvider(curationFiles: Collection<File>) : PackageCura
             }.onFailure {
                 log.warn { "Failed parsing package curation from '${curationFile.absoluteFile}'." }
             }.getOrNull()
-        }.flatten()
+        }.flatten().toSet()
```

Comment:
I believe this change may have an effect on the result of applying the curations. It wouldn't have an effect if the ordering wasn't effected and always the foremost of the duplicated curations were removed. I doubt this is the case.

Apart from that, is it deterministic still?

I'm not 100% convinced that this is the right way to address the problem. 
What do you think about the following two approaches
- throw an exception instead
- accept duplicates, but address it elsewhere. Maybe have a duplicates check in your PSV.

I'd be interested in what other devs think about this - maybe @oss-review-toolkit/core-devs ?

Reply:
As I can't see a valid use case to use contradicting duplicated curations, I think explicitly throwing an exception would be the best way to solve this.

Reply:
Should we then also throw an exception when the parsing fails (instead of just logging a warning) ? 

..and probably also ditch the `by lazy` in order to fail early?

Reply:
> Should we then also throw an exception when the parsing fails

This was suggested by @sschuberth here: https://github.com/oss-review-toolkit/ort/pull/4131#discussion_r648273567
For our internal workflow, our CI checks if the curations can be deserialised anyways.

Does it make sense to support use cases where the user uses a `curations/` directory with other `.yml` and `.json` files (and just print a warning)?
Otherwise I'd suggest to throw an exception and fail the ORT run.

Reply:
For now I just added a check and the `Exception` if curations are duplicated.

Reply:
> Does it make sense to support use cases where the user uses a curations/ directory with other .yml and .json files (and just print a warning)?
Otherwise I'd suggest to throw an exception and fail the ORT run.

I believe it makes sense to assume that underneath the given package curations directory all files are package curation files, because it's simpler and its a requirement one can IMO easily ask for. It does not harm any user if it can't put other files there.